### PR TITLE
Implement getNearbyEntities

### PR DESCRIPTION
**Pull request by nothub**
# Description
This PR implements `getNearbyEntities` in 4 variations.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [x] Unit tests added (if applicable).

### CODE REVIEW

**Review by Insprill — CHANGES_REQUESTED** [review]
Looks good, but please add unit tests.

**Comment by nothub** [comment]
I added 2 Tests, one for testing the distance filter and one for testing the predicate filter.

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java [hunk 1]
```diff
@@ -2245,22 +2244,22 @@ public boolean isChunkGenerated(int x, int z)
...
+		return getNearbyEntities(BoundingBox.of(location, x, y, z), filter);
 	}
 
 	@Override
 	public @NotNull Collection<Entity> getNearbyEntities(BoundingBox boundingBox)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return getNearbyEntities(boundingBox, null);
 	}
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/WorldMock.java:2254]: Please add a Test for this

**Review by thelooter — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java (hunk 1 above)
Comment (nothub) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java:2254]: Tests added for BoundingBox methods.

**Review by thelooter — COMMENTED** [review]
LGTM

**Review by thelooter — APPROVED** [review]

**Review by Insprill — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**